miraclelinux/meta-debian-extended
# vim: Update version from 8.1.0875-5+deb10u4 to 8.1.0875-5+deb10u5

# Purpose of pull request

This version fixes following CVEs.
- CVE-2022-4141
- CVE-2023-0054
- CVE-2023-1175
- CVE-2023-2610

DLA: DLA 3453-1

# Test
1. Build package
2. Run image and test the package on the qemuarm64 environment

## How to test the package
1. Build qemuarm64 image
2. Run qemu image
3. Run commands check for vim is working

### Build qemuarm64 image
Add the following to local.conf and build qemuarm64 image.
```
IMAGE_INSTALL_append = " vim"
```
### Run qemu image
```
$ runqemu qemuarm64 nographic
```
### Run commands check for vim is working
```
# ls -la /bin/vi
# vi test.txt
# cat test.txt
# cat test.txt | vi -es +'%s/BBB/DDD' +'%p' +'q!' /dev/stdin
```
## Test result
### build
Build succeeded.
```
$ bitbake vim
Checking Debian Release ...
Sources.xz is not changed.
Force regenerate source code information.
Parsing Debian Sources.xz ...
Parsing recipes: 100% |##########################################################################################################################################| Time: 0:00:46
Parsing of 1362 .bb files complete (0 cached, 1362 parsed). 2384 targets, 80 skipped, 6 masked, 0 errors.
NOTE: Resolving any missing task queue dependencies

Build Configuration:
BB_VERSION           = "1.42.0"
BUILD_SYS            = "x86_64-linux"
NATIVELSBSTRING      = "ubuntu-20.04"
TARGET_SYS           = "aarch64-emlinux-linux"
MACHINE              = "qemuarm64"
DISTRO               = "emlinux"
DISTRO_VERSION       = "2.7"
TUNE_FEATURES        = "aarch64 armv8a crc"
TARGET_FPU           = ""
meta                 
meta-yocto-bsp       = "HEAD:d4b57c68b22027c2bedff335dee06af963e4f8a8"
meta-debian          = "HEAD:b622b264446afaab533290ceb59ad84096caedc7"
meta-debian-extended = "update-vim_8.1.0875-5+deb10u5:b36690033a268804a9293610de194b19fa629862"
meta-emlinux         = "HEAD:274ff063532e8d251a5a5a71fc7e2dbdf3a6dac8"
meta-emlinux-private = "HEAD:8207525289999d2c41983eb8155127b62621cdb1"

Initialising tasks: 100% |#######################################################################################################################################| Time: 0:00:06
Sstate summary: Wanted 624 Found 0 Missed 624 Current 0 (0% match, 0% complete)
NOTE: Executing SetScene Tasks
NOTE: Executing RunQueue Tasks
NOTE: Tasks Summary: Attempted 2909 tasks of which 0 didn't need to be rerun and all succeeded.
```

### Package test
Edit file via vi.
```
root@qemuarm64:~# ls -la /bin/vi
lrwxrwxrwx    1 root     root            16 Jun 16 10:07 /bin/vi -> /usr/bin/vim.vim
root@qemuarm64:~# vi test.txt
root@qemuarm64:~# cat test.txt
AAA
BBB
CCC
root@qemuarm64:~# cat test.txt | vi -es +'%s/BBB/DDD' +'%p' +'q!' /dev/stdin
AAA
DDD
CCC
```
